### PR TITLE
Added Year into groupping criteria

### LINKED ISSUE #29: Schedule order over several years
When schedule spans over several years schedule days should be grouped by year, then by week, then by day and so on

### PR DESCRIPTION
**Pull request by sdiachen**
closes #29

### CODE REVIEW

File: VolleyManagement.UI/Areas/WebAPI/Controllers/TournamentsController.cs [hunk 1]
```diff
@@ -107,20 +107,14 @@ public List<PivotStandingsViewModel> GetTournamentPivotStandings(int id)
...
-                gr.Date, CalendarWeekRule.FirstDay, DayOfWeek.Monday))
-                .OrderBy(w => w.Key)
-                .Select(w => new Tuple<int, List<GameViewModel>>(w.Key, w.ToList()))
+            var resultGroupedByWeek = games.GroupBy(GetWeekOfYear)
+                .OrderBy(w => w.Key.Year)
+                .ThenBy(w => w.Key.Week)
+                .Select(w => new Tuple<int, List<GameViewModel>>(w.Key.Week, w.ToList()))
                 .ToList();
 
             var result = new ScheduleViewModel()
```
Comment (dmshapoval) [inline VolleyManagement.UI/Areas/WebAPI/Controllers/TournamentsController.cs]: Did you think about making this query more readable?

File: VolleyManagement.UI/Areas/WebAPI/Controllers/TournamentsController.cs [hunk 2]
```diff
@@ -107,20 +107,14 @@ public List<PivotStandingsViewModel> GetTournamentPivotStandings(int id)
...
-            List<GameViewModel> gamesViewModel = _gameService.GetTournamentGames(tournamentId)
-                                                        .Select(t => GameViewModel.Map(t)).ToList();
-            foreach (var item in gamesViewModel)
-            {
-                if (item.Result.TotalScore.IsEmpty)
-                {
-                    item.Result = null;
-                }
-            }
+            var games = _gameService.GetTournamentGames(tournamentId)
```
Comment (dmshapoval) [inline VolleyManagement.UI/Areas/WebAPI/Controllers/TournamentsController.cs]: It seems for me we don't need invoke ToList()

File: VolleyManagement.UnitTests/Services/GameService/Fixture/GameServiceTestFixture.cs [hunk 3]
```diff
@@ -562,14 +557,14 @@ public GameServiceTestFixture TestGamesWithResultInThreeWeeksTwoDivisionsThreeGa
...
-                    GameScore = new Score(1, 3),
+                    GameScore = (1, 3),
                     SetScores = new List<Score>
                     {
-                        new Score(25, 0),
-                        new Score(5, 25),
-                        new Score(5, 25),
-                        new Score(5, 25),
-                        new Score(0, 0),
+                        (25, 0),
```
Comment (dmshapoval) [inline VolleyManagement.UnitTests/Services/GameService/Fixture/GameServiceTestFixture.cs]: Agree, it looks better)

File: VolleyManagement.UnitTests/WebApi/ViewModels/Schedule/ScheduleViewModelTestFixture.cs [hunk 4]
```diff
@@ -558,6 +560,167 @@ public ScheduleViewModelTestFixture WithThreeWeeksTwoDivisionsThreeGames()
...
+                            {
+                                new GameViewModel
+                                {
+                                    Id = 1,
+                                    HomeTeamName = "TeamNameA",
+                                    AwayTeamName = "TeamNameB",
+                                    GameDate = DateTime.Parse(DATE_A).ToString("yyyy-MM-ddTHH:mm:sszzz"),
+                                    Result = new GameViewModel.GameResult
+                                    {
+                                        TotalScore = new UI.Areas.Mvc.ViewModels.GameResults.ScoreViewModel(3, 0),
```
Comment (dmshapoval) [inline VolleyManagement.UnitTests/WebApi/ViewModels/Schedule/ScheduleViewModelTestFixture.cs]: usings?

File: VolleyManagement.UI/Areas/WebAPI/Controllers/TournamentsController.cs (hunk 1 above)
Comment (sdiachen) [reply VolleyManagement.UI/Areas/WebAPI/Controllers/TournamentsController.cs]: See PR #45

File: VolleyManagement.UnitTests/WebApi/ViewModels/Schedule/ScheduleViewModelTestFixture.cs (hunk 4 above)
Comment (sdiachen) [reply VolleyManagement.UnitTests/WebApi/ViewModels/Schedule/ScheduleViewModelTestFixture.cs]: It was a copy paste from another place. I believe we have ScoreViewModel in API area as well and to remove ambiguity

File: VolleyManagement.UI/Areas/WebAPI/Controllers/TournamentsController.cs (hunk 2 above)
Comment (sdiachen) [reply VolleyManagement.UI/Areas/WebAPI/Controllers/TournamentsController.cs]: I'll fix that

**Pull request merged**